src/apint/arithmetic.rs

```diff
@@ -787,75 +1676,58 @@ mod tests {
-                            if temp != 0 {
-                                panic!(
-                                    "\nlhs_nine:{}\nrhs_nine:{}\nresult:{:?}\nzeros_after_one:{}\nnines_before_eight:{}\nnines_after_eight:{}\n",
-                                    lhs_nine, rhs_nine, result,zeros_after_one,nines_before_eight,nines_after_eight
-                                );
+                                continue 'outer
                             }
+                            result.wrapping_lshr_assign(8usize).unwrap();
+                            assert_eq!(result.clone().resize_to_u8(),0);
                         }
                         for i in 0..nines_before_eight {
                             if zeros_after_one + i >= num_u8 - 1 {
-                                continue 'outer;
+                                continue 'outer
                             }
-                            result.wrapping_lshr_assign(8).unwrap();
+                            result.wrapping_lshr_assign(8usize).unwrap();
                             assert_eq!(result.clone().resize_to_u8(), u8::MAX);
                         }
                         if zeros_after_one + nines_before_eight >= num_u8 - 1 {
-                            continue 'outer;
-                        }
-                        result.wrapping_lshr_assign(8).unwrap();
-                        let temp = result.clone().resize_to_u8();
-                        if temp != u8::MAX - 1 {
-                            panic!(
-                                "\nlhs_nine:{}\nrhs_nine:{}\nresult:{:?}\nzeros_after_one:{}\nnines_before_eight:{}\nnines_after_eight:{}\n",
-                                lhs_nine, rhs_nine, result,zeros_after_one,nines_before_eight,nines_after_eight
-                            );
+                            continue 'outer
                         }
+                        result.wrapping_lshr_assign(8usize).unwrap();
+                        assert_eq!(result.clone().resize_to_u8(),u8::MAX - 1);
                         for i in 0..nines_after_eight {
                             if 1 + zeros_after_one + nines_before_eight + i >= num_u8 - 1 {
-                                continue 'outer;
-                            }
-                            result.wrapping_lshr_assign(8).unwrap();
-                            if result.clone().resize_to_u8() != u8::MAX {
-                                panic!(
-                                    "\nlhs_nine:{}\nrhs_nine:{}\nresult:{:?}\nzeros_after_one:{}\nnines_before_eight:{}\nnines_after_eight:{}\n",
-                                    lhs_nine, rhs_nine, result,zeros_after_one,nines_before_eight,nines_after_eight
-                                );
+                                continue 'outer
                             }
+                            result.wrapping_lshr_assign(8usize).unwrap();
+                            assert_eq!(result.clone().resize_to_u8(),u8::MAX);
```

Comment:
`result.clone()` seems to be a very inefficient operation. So maybe we should use `debug_assert_eq!` here instead of a normal `assert_eq!` to avoid inefficiencies in release code.

Reply:
And I think the same could and should be applied to all other applicable cases in this code of course.

Reply:
Do you realize that is testing code? It has to be `assert_eq` because I compile with `cargo test --release`. Also, I should note that running all the tests requires `--release` now because of the fuzz testers, or else it will take a long time. At the very bottom of `megafuzz` there are commented out lines that take something like an hour to run so I only run them whenever I am going to issue a pull request that modifies arithmetic functions. The `.clone` I am using everywhere in the tests are needed because I need to test one thing against many things.